cmd/appliance/seed.go

```diff
@@ -0,0 +1,276 @@
+					}
+					password, err := prompt.GetPassphrase(opts.In, opts.CanPrompt, hasStdin, "Appliance's CZ user password:")
+					if err != nil {
+						return err
+					}
+					opts.password = password
+				}
+
+			case 1:
+				if !util.IsUUID(args[0]) {
+					return errors.New("expected appliance UUID")
+				}
+				opts.applianceID = args[0]
+
+			}
+
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			return seedRun(c, args, &opts)
+		},
+	}
+	cmd.Flags().BoolVar(&opts.provideCloudSSHKey, "provide-cloud-ssh-key", false, "Tells appliance to use the key generated by AWS or Azure")
+	cmd.Flags().BoolVar(&opts.allowCustomization, "allow-customization", false, "Whether the Appliance should allow customizations or not")
+	cmd.Flags().BoolVar(&opts.iso, "iso-format", false, "Export as ISO format")
+	cmd.Flags().BoolVar(&opts.json, "json", false, "output as json to stdout, only applicable in json seed type")
+	cmd.Flags().StringVarP(&opts.sshKey, "ssh-key", "", "", "filepath to public ssh key to be allowed")
+	cmd.Flags().IntVarP(&opts.validityDays, "validity-days", "", 1, "How many days the seed should be valid for")
+
+	return cmd
+}
+
+func seedRun(cmd *cobra.Command, args []string, opts *seedOpts) error {
+	cfg := opts.Config
+	a, err := opts.Appliance(cfg)
+	if err != nil {
+		return err
+	}
+	token, err := opts.Config.GetBearTokenHeaderValue()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	appliance, err := a.Get(ctx, opts.applianceID)
+	if err != nil {
+		return err
+	}
+	if appliance.GetActivated() {
+		return fmt.Errorf("appliance %s is already activated", appliance.GetName())
+	}
+
+	sshConfig := openapi.NewSSHConfig()
+
+	if cfg.Version < 15 {
+		sshConfig.AllowCustomization = nil
+		sshConfig.ValidityDays = nil
+	}
+	sshConfig.AllowCustomization = openapi.PtrBool(opts.allowCustomization)
+	sshConfig.ValidityDays = openapi.PtrFloat32(float32(opts.validityDays))
+
+	if len(opts.sshKey) > 0 {
+		pub, err := readFile(opts.sshKey)
+		if err != nil {
+			return err
+		}
+		sshConfig.SetSshKey(pub)
+	} else if opts.provideCloudSSHKey {
+		sshConfig.SetProvideCloudSSHKey(true)
+	} else if len(opts.password) < 1 {
+		hasStdin := false
+		stat, err := os.Stdin.Stat()
+		if err == nil && (stat.Mode()&os.ModeCharDevice) == 0 {
+			hasStdin = true
+		}
+		password, err := prompt.GetPassphrase(opts.In, false, hasStdin, "")
+		if err != nil {
+			return errors.New("You need to provide passphrase through stdin or select --ssh-key=/path/to/key.pub or --provide-cloud-ssh-key")
+		}
+		sshConfig.SetPassword(password)
+	} else {
+		sshConfig.SetPassword(opts.password)
+	}
+
+	var (
+		data []byte
+		ext  string
+	)
+	if opts.iso {
+		seed, response, err := a.APIClient.AppliancesApi.AppliancesIdExportIsoPost(ctx, appliance.GetId()).SSHConfig(*sshConfig).Authorization(token).Execute()
+		if err != nil {
+			return api.HTTPErrorResponse(response, err)
+		}
+		data = []byte(seed.GetIso())
+		ext = "iso"
+	} else {
```

Comment:
Since there are flags for both json and iso, I think we should explicitly check for the json flag value instead. This logic implies that json is the default format, which would make the json flag obsolete.